roomserver/storage/shared/storage.go

```diff
@@ -701,6 +701,31 @@ func (d *Database) GetOrCreateRoomInfo(ctx context.Context, event gomatrixserver
 	}, err
 }
 
+// GetOrCreateRoomInfo gets or creates a new RoomInfo, which is only safe to use with functions only needing a roomVersion or roomNID.
+func (d *Database) GetOrCreateRoomInfoFromID(ctx context.Context, roomID string) (roomInfo *types.RoomInfo, err error) {
```

Comment:
Name is misleading, or missing something in the function body.
We don't assign a new roomNID if `roomInfo` is nil.